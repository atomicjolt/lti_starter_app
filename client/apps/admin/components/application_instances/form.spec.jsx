import React from 'react';
import { shallow } from 'enzyme';
import Form from './form';

describe('application instance form', () => {
  let result;
  let props;
  let modalClosed = false;
  let saved;

  beforeEach(() => {
    saved = false;
    props = {
      onChange: () => {},
      closeModal: () => { modalClosed = true; },
      save: () => { saved = true; },
      newSite: () => {},
      site_id: 'foo',
      sites: {},
      config: '{ "foo": "bar" }',
      ltiConfigParseError: '',
    };
    result = shallow(<Form {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the close modal event', () => {
    expect(modalClosed).toBeFalsy();
    result.find('.c-btn--gray--large').simulate('click');
    expect(modalClosed).toBeTruthy();
  });

  it('handles the save event', () => {
    expect(saved).toBeFalsy();
    result.find('.c-btn--yellow').simulate('click');
    expect(saved).toBeTruthy();
  });
});
