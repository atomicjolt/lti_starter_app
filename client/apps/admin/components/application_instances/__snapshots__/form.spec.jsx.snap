// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application instance form matches the snapshot 1`] = `
<form>
  <div
    className="o-grid o-grid__modal-top"
  >
    <div
      className="o-grid__item u-half"
    >
      <Input
        className="c-input"
        inputProps={
          Object {
            "id": "nickname_input",
            "name": "nickname",
            "onChange": [Function],
            "type": "text",
            "value": undefined,
          }
        }
        labelText="Nickname"
      />
    </div>
    <div
      className="o-grid__item u-half"
    >
      <Input
        className="c-input"
        inputProps={
          Object {
            "id": "primary_contact_input",
            "name": "primary_contact",
            "onChange": [Function],
            "type": "text",
            "value": undefined,
          }
        }
        labelText="Primary contact"
      />
    </div>
    <div
      className="o-grid__item u-half"
    >
      <div
        className="c-input"
      >
        <span>
          Canvas Url
        </span>
        <StateManager
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          isClearable={false}
          name="site_id"
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": <div>
                  Add New
                </div>,
                "onSelect": [Function],
                "value": "new",
              },
            ]
          }
          placeholder="Select a Canvas Domain"
        />
      </div>
    </div>
    <div
      className="o-grid__item u-half"
    >
      <Input
        className="c-input"
        inputProps={
          Object {
            "id": "domain_input",
            "name": "domain",
            "onChange": [Function],
            "type": "text",
            "value": undefined,
          }
        }
        labelText="LTI Tool Domain"
      />
    </div>
    <div
      className="o-grid__item u-half"
    >
      <Input
        className="c-input"
        inputProps={
          Object {
            "disabled": undefined,
            "id": "lti_key_input",
            "name": "lti_key",
            "onChange": [Function],
            "type": "text",
            "value": undefined,
          }
        }
        labelText="LTI Key"
      />
    </div>
    <div
      className="o-grid__item u-half"
    >
      <Input
        className="c-input"
        inputProps={
          Object {
            "id": "lti_secret_input",
            "name": "lti_secret",
            "onChange": [Function],
            "type": "text",
            "value": undefined,
          }
        }
        labelText="LTI Secret"
      />
    </div>
    <div
      className="o-grid__item u-half"
    >
      <Input
        className="c-input"
        helperText="Current Canvas Token: undefined"
        inputProps={
          Object {
            "id": "canvas_token_input",
            "name": "canvas_token",
            "onChange": [Function],
            "placeholder": "",
            "type": "text",
            "value": "",
          }
        }
        labelText="Canvas Token"
      />
    </div>
    <div
      className="o-grid__item u-full"
    >
      <Input
        className="c-checkbox"
        helperText="Indicates whether or not user name and email is stored during LTI launch"
        inputProps={
          Object {
            "checked": undefined,
            "id": "anonymous_input",
            "name": "anonymous",
            "onChange": [Function],
            "type": "checkbox",
            "value": "true",
          }
        }
        labelText="Anonymous"
      />
    </div>
    <div
      className="o-grid__item u-full"
    >
      <Input
        className="c-checkbox"
        helperText="Indicates whether or not rollbar is enabled for this app instance"
        inputProps={
          Object {
            "checked": undefined,
            "id": "rollbar_enabled_input",
            "name": "rollbar_enabled",
            "onChange": [Function],
            "type": "checkbox",
            "value": "true",
          }
        }
        labelText="Rollbar Enabled"
      />
    </div>
    <div
      className="o-grid__item u-full"
    >
      <Input
        className="c-checkbox"
        helperText="Indicates this is a paid or trial account"
        inputProps={
          Object {
            "checked": false,
            "id": "paid_input",
            "name": "paid",
            "onChange": [Function],
            "type": "checkbox",
            "value": "true",
          }
        }
        labelText="Paid Account"
      />
    </div>
    <div
      className="o-grid__item u-full"
    >
      <Input
        className="c-checkbox"
        helperText="Restricts the Canvas tokens generated during oauth to the minimum necessary for this application. This should only be used if the oauth key and secret are populated above and are for a scoped developer key."
        inputProps={
          Object {
            "checked": undefined,
            "id": "use_scoped_developer_key_input",
            "name": "use_scoped_developer_key",
            "onChange": [Function],
            "type": "checkbox",
            "value": "true",
          }
        }
        labelText="Use Scoped Developer Key"
      />
    </div>
    <div
      className="o-grid__item u-half"
    >
      <div
        className="c-input c-input--container"
      >
        <span>
          Language
        </span>
        <StateManager
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          isClearable={false}
          name="language"
          onChange={[Function]}
          options={Array []}
          placeholder="english"
        />
      </div>
    </div>
    <div
      className="o-grid__item u-full"
    >
      <Textarea
        className="c-input"
        labelText="Custom Application Instance Configuration"
        textareaProps={
          Object {
            "id": "application_instance_config",
            "name": "config",
            "onChange": [Function],
            "placeholder": "ex: { \\"foo\\": \\"bar\\" }",
            "rows": 3,
            "value": "{
  \\"foo\\": \\"bar\\"
}",
          }
        }
        warning={null}
      />
    </div>
    <div
      className="o-grid__item u-full"
    >
      <Textarea
        className="c-input"
        labelText="LTI Configuration"
        textareaProps={
          Object {
            "id": "application_instance_lti_config",
            "name": "lti_config",
            "onChange": [Function],
            "rows": 8,
            "value": "{}",
          }
        }
        warning={null}
      />
    </div>
  </div>
  <button
    className="c-btn c-btn--yellow"
    onClick={[Function]}
    type="button"
  >
    Save
  </button>
  <button
    className="c-btn c-btn--gray--large u-m-right"
    onClick={[Function]}
    type="button"
  >
    Cancel
  </button>
</form>
`;
